### PR TITLE
System pathways updates

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: app/assets/javascripts/chart_loader.es6 [hunk 1]
```diff
@@ -116,28 +116,31 @@ App.StimulusApp.register('chart-loader', class extends Stimulus.Controller {
...
   }
 
+  // Takes an event or a jQuery object representing the link
   loadChartData(event) {
-    event.preventDefault();
+    if (event.target) event.preventDefault();
     this.enableLoader();
     this.hideChartAndTable();
-    let url = this.activeTarget(event).href;
+    let target = target = this.activeTarget(event)
```
Comment (awisema) [inline app/assets/javascripts/chart_loader.es6]: I think this should just be:
```suggestion
    let target = this.activeTarget(event);
```

File: drivers/system_pathways/app/models/system_pathways/chart_base.rb [hunk 2]
```diff
@@ -181,9 +180,16 @@ def average(value, count)
...
+      scope = scope.where(involves_ce: true) if details_filter&.involves_ce == 'Yes'
+      scope = scope.where(involves_ce: false) if filter.involves_ce == 'No'
+      scope = scope.where(involves_ce: false) if show_filter&.involves_ce == 'No'
+      scope = scope.where(involves_ce: false) if details_filter&.involves_ce == 'No'
+      scope = scope.where(ce_assessment: true) if filter.involves_ce == 'With CE Assessment'
+      scope = scope.where(ce_assessment: true) if show_filter&.involves_ce == 'With CE Assessment'
+      scope = scope.where(ce_assessment: true) if details_filter&.involves_ce == 'With CE Assessment'
+
       scope
     end
```
Comment (awisema) [inline drivers/system_pathways/app/models/system_pathways/chart_base.rb:194]:
```ruby
def involves_ce?(mode)
[:filter, :show_filter, :details_filter].map do |which_filter|
  send(which_filter) == mode
end.any?
```

File: drivers/system_pathways/app/models/system_pathways/chart_base.rb [hunk 3]
```diff
@@ -181,9 +180,16 @@ def average(value, count)
...
+      scope = scope.where(involves_ce: true) if filter.involves_ce == 'Yes'
+      scope = scope.where(involves_ce: true) if show_filter&.involves_ce == 'Yes'
+      scope = scope.where(involves_ce: true) if details_filter&.involves_ce == 'Yes'
+      scope = scope.where(involves_ce: false) if filter.involves_ce == 'No'
+      scope = scope.where(involves_ce: false) if show_filter&.involves_ce == 'No'
+      scope = scope.where(involves_ce: false) if details_filter&.involves_ce == 'No'
+      scope = scope.where(ce_assessment: true) if filter.involves_ce == 'With CE Assessment'
+      scope = scope.where(ce_assessment: true) if show_filter&.involves_ce == 'With CE Assessment'
+      scope = scope.where(ce_assessment: true) if details_filter&.involves_ce == 'With CE Assessment'
+
```
Comment (awisema) [inline drivers/system_pathways/app/models/system_pathways/chart_base.rb:192]:
```suggestion
      scope = scope.where(involves_ce: true) if involves_ce?('Yes')
      scope = scope.where(involves_ce: false) if involves_ce?('No')
      scope = scope.where(ce_assessment: true) if involves_ce?(With CE Assessment')
      
```

File: drivers/system_pathways/app/views/system_pathways/warehouse_reports/reports/_pathways_chart.haml [hunk 4]
```diff
@@ -7,6 +7,15 @@
...
   #sankey
+%p The above chart tracks clients with open homeless enrollments in the selected universe.  System pathways are limited to those that proceed through the continuum.  If a client goes backwards within the continuum (from more housed to less housed) the regression is ignored for the puposes of this report unless they first exited to a permanent destination.  Clients who exited to a permanent destination and then subsequently enrolled in a homeless project are counted in the Returns to Homelessness category.  The progression of least to most homeless is as follows:
+%ul
+  %li SO (4)
+  %li ES (1)
+  %li SH (8)
+  %li TH (2)
+  %li PH - RRH (13)
+  %li PH - PSH (3), PH - PH (9), PH - OPH (10)
+
```
Comment (awisema) [inline drivers/system_pathways/app/views/system_pathways/warehouse_reports/reports/_pathways_chart.haml]: I don't understand what 'least to most homeless` means, and how it relates to the list...

**Review by awisema — CHANGES_REQUESTED** [review]

File: drivers/system_pathways/app/models/system_pathways/chart_base.rb (hunk 2 above)
Comment (awisema) [reply drivers/system_pathways/app/models/system_pathways/chart_base.rb:194]: This goes with the next conversation.

File: drivers/system_pathways/app/models/system_pathways/chart_base.rb (hunk 3 above)
Comment (eanders) [reply drivers/system_pathways/app/models/system_pathways/chart_base.rb:192]: This is actually correct as written.

**Pull request merged**